extensions/pkg/webhook/utils.go

```diff
@@ -156,128 +179,133 @@ func EnsureNoStringWithPrefixContains(items []string, prefix, value, sep string)
-	if i := vpaContainerResourcePolicyWithNameIndex(items, item.ContainerName); i < 0 {
-		items = append(items, item)
-	} else if !reflect.DeepEqual(items[i], item) {
-		items = append(append(items[:i], item), items[i+1:]...)
+	i := slices.IndexFunc(items, func(crp vpaautoscalingv1.ContainerResourcePolicy) bool {
+		return crp.ContainerName == item.ContainerName
+	})
+	if i < 0 {
+		return append(items, item)
 	}
-	return items
+	return append(append(items[:i], item), items[i+1:]...)
 }
 
 // EnsurePVCWithName ensures that a PVC with a name equal to the name of the given PVC exists
-// in the given slice and is equal to the given PVC.
+// in the given slice and the first item in the list would be equal to the given PVC.
 func EnsurePVCWithName(items []corev1.PersistentVolumeClaim, item corev1.PersistentVolumeClaim) []corev1.PersistentVolumeClaim {
-	if i := pvcWithNameIndex(items, item.Name); i < 0 {
-		items = append(items, item)
-	} else if !reflect.DeepEqual(items[i], item) {
-		items = append(append(items[:i], item), items[i+1:]...)
+	i := slices.IndexFunc(items, func(pvc corev1.PersistentVolumeClaim) bool {
+		return pvc.Name == item.Name
+	})
+	if i < 0 {
+		return append(items, item)
 	}
-	return items
+	return append(append(items[:i], item), items[i+1:]...)
 }
 
 // EnsureNoPVCWithName ensures that a PVC with the given name does not exist in the given slice.
 func EnsureNoPVCWithName(items []corev1.PersistentVolumeClaim, name string) []corev1.PersistentVolumeClaim {
-	if i := pvcWithNameIndex(items, name); i >= 0 {
-		items = append(items[:i], items[i+1:]...)
-	}
-	return items
+	return slices.DeleteFunc(items, func(pvc corev1.PersistentVolumeClaim) bool {
+		return pvc.Name == name
+	})
 }
 
 // EnsureUnitOption ensures the given unit option exist in the given slice.
 func EnsureUnitOption(items []*unit.UnitOption, item *unit.UnitOption) []*unit.UnitOption {
-	if i := unitOptionIndex(items, item); i < 0 {
-		items = append(items, item)
+	i := slices.IndexFunc(items, func(uo *unit.UnitOption) bool {
+		return reflect.DeepEqual(uo, item)
+	})
+	if i < 0 {
+		return append(items, item)
 	}
 	return items
 }
 
 // EnsureFileWithPath ensures that a file with a path equal to the path of the given file exists in the given slice
 // and is equal to the given file.
 func EnsureFileWithPath(items []extensionsv1alpha1.File, item extensionsv1alpha1.File) []extensionsv1alpha1.File {
-	if i := fileWithPathIndex(items, item.Path); i < 0 {
-		items = append(items, item)
-	} else if !reflect.DeepEqual(items[i], item) {
-		items = append(append(items[:i], item), items[i+1:]...)
+	i := slices.IndexFunc(items, func(f extensionsv1alpha1.File) bool {
+		return f.Path == item.Path
+	})
+	if i < 0 {
+		return append(items, item)
 	}
-	return items
+	return append(append(items[:i], item), items[i+1:]...)
```

Comment:
Can't we do something like
```suggestion
        items[i] = item
	return items
```

Reply:
Yes that way it would be cleaner but won't this modify the parameter array? The previous code does not modify it

Reply:
I am not sure that I get what you mean. The previous code was doing 
```golang
items = append(append(items[:i], item), items[i+1:]...)
```

Reply:
Anyway after testing `append(append(items[:i], item), items[i+1:]...)` and `items[i]=item` seem to have the same side effect